app/main/updater.js

```diff
@@ -0,0 +1,56 @@
+import {dialog} from 'electron';
+import {autoUpdater} from 'electron-updater';
+
+import {t} from './helpers';
+
+const RELEASES_LINK = 'https://github.com/appium/appium-inspector/releases';
+
+autoUpdater.autoDownload = false;
+autoUpdater.autoInstallOnAppQuit = false;
+
+autoUpdater.on('error', (error) => {
+  dialog.showErrorBox(t('Could not download update'), t('updateDownloadFailed', {message: error}));
+});
+
+autoUpdater.on('update-not-available', () => {
+  dialog.showMessageBox({
+    type: 'info',
+    buttons: [t('OK')],
+    message: t('No update available'),
+    detail: t('Appium Inspector is up-to-date'),
+  });
+});
+
+autoUpdater.on('update-available', async ({version, releaseDate}) => {
+  const pubDate = new Date(releaseDate).toDateString();
+  const {response} = await dialog.showMessageBox({
+    type: 'info',
+    message: t('appiumIsAvailable', {name: version}),
+    buttons: [t('Install Now'), t('Install Later')],
+    detail: t('updateDetails', {pubDate, notes: RELEASES_LINK}),
+  });
+  if (response === 0) {
+    dialog.showMessageBox({
+      type: 'info',
+      buttons: [t('OK')],
+      message: t('updateIsBeingDownloaded'),
+    });
+    autoUpdater.downloadUpdate();
```

Comment:
is this done silently in background?

Reply:
Yes. However, unlike stock Electron, `electron-updater` has the `download-progress` event, which could theoretically be used to show the progress percentage. But from a quick glance, it would require creating a dedicated window and sending the progress text as a message event to this window. I think it's still doable, but perhaps not as critical to have at the moment.

Reply:
makes sense